lib/shelley/src/Cardano/Wallet/Shelley/Network/Blockfrost.hs

```diff
@@ -387,58 +393,65 @@ instance FromBlockfrost BF.Epoch EpochNo where
+│  236  │ Allegra │   │  102  │ Allegra │
+├───────┼─────────┤   ├───────┼─────────┤
+│  235  │ Shelley │   │  101  │ Shelley │
+│  ...  │ Shelley │   │  ...  │ Shelley │
+│  208  │ Shelley │   │   74  │ Shelley │
+├───────┼─────────┤   ├───────┼─────────┤
+│  207  │  Byron  │   │   73  │  Byron  │
+│  ...  │  Byron  │   │  ...  │  Byron  │
+│    0  │  Byron  │   │    0  │  Byron  │
+└───────┴─────────┘   └───────┴─────────┘
+
 -}
 eraByEpoch :: NetworkId -> EpochNo -> Either BlockfrostError AnyCardanoEra
-eraByEpoch = \case
-    Mainnet -> \epoch ->
-        case dropWhile ((> epoch) . snd) (reverse eraBoundaries) of
-            (era, _) : _ -> Right era
-            _ -> Left $ UnknownEraForEpoch epoch
-    Testnet _ -> \_ -> error
-        "In light-mode era to epoch conversions are only available for the \
-        \mainnet (translation uses a hard-coded history of hard forks). \
-        \It doesn't seem viable to hardcode eras for other networks yet."
-
-eraBoundaries :: [(Node.AnyCardanoEra, EpochNo)]
-eraBoundaries = [minBound .. maxBound] <&> \era -> (era, epochEraStartsAt era)
+eraByEpoch net epoch =
+    case dropWhile ((> epoch) . snd) (reverse eraBoundaries) of
+        (era, _) : _ -> Right era
+        _ -> Left $ UnknownEraForEpoch epoch
   where
-    -- When new era is added this function reminds to update itself:
-    -- "Pattern match(es) are non-exhaustive"
-    epochEraStartsAt :: Node.AnyCardanoEra -> EpochNo
-    epochEraStartsAt = EpochNo . \case
-        Node.AnyCardanoEra Node.AlonzoEra  -> 290
-        Node.AnyCardanoEra Node.MaryEra    -> 251
-        Node.AnyCardanoEra Node.AllegraEra -> 236
-        Node.AnyCardanoEra Node.ShelleyEra -> 202
-        Node.AnyCardanoEra Node.ByronEra   -> 0
+    eraBoundaries :: [(Node.AnyCardanoEra, EpochNo)]
+    eraBoundaries =
+        [minBound .. maxBound] <&> \era -> (era, epochEraStartsAt era)
+      where
+        -- When new era is added this function reminds to update itself:
+        -- "Pattern match(es) are non-exhaustive"
+        epochEraStartsAt :: Node.AnyCardanoEra -> EpochNo
+        epochEraStartsAt era = EpochNo $ case net of
+            Mainnet ->
+                case era of
+                    AnyCardanoEra AlonzoEra  -> 290
+                    AnyCardanoEra MaryEra    -> 251
+                    AnyCardanoEra AllegraEra -> 236
+                    AnyCardanoEra ShelleyEra -> 208
+                    AnyCardanoEra ByronEra   -> 0
+            Testnet (NetworkMagic 1097911063) ->
+                case era of
+                    AnyCardanoEra AlonzoEra  -> 154
+                    AnyCardanoEra MaryEra    -> 112
+                    AnyCardanoEra AllegraEra -> 102
+                    AnyCardanoEra ShelleyEra -> 74
+                    AnyCardanoEra ByronEra   -> 0
+            Testnet magic ->
+                error $ "Epoch/Era conversion isn't provided for the Testnet "
+                    <> show magic <> " in light mode."
```

Comment:
Since with Blockfrost provider we are supporting only mainnet and public testnet (i.e. the one with protocolMagic -> 1097911063) then perhaps we shouldn't be able to start wallet against any different testnet alltogether. E.g.:

If we start wallet with:
```
cardano-wallet serve \
  --light \
  --blockfrost-token-file blockfrost.api.key \
  --testnet testnet-byron-genesis.json
```
and `testnet-byron-genesis.json` contains protocolMagic different than `1097911063` the wallet should just stop with the appropriate message.

_Maybe it's not in scope of this pr though_

Reply:
Or maybe starting against public testnet could be simple as:
```
cardano-wallet serve \
  --light \
  --blockfrost-token-file blockfrost.api.key \
  --testnet
```
i.e. without the byron-genesis.json, assuming it is needed only for getting the protocolMagic from.